sci-mathematics/calc/calc-2.12.5.3.recipe

```diff
@@ -0,0 +1,117 @@
+SUMMARY="C-style arbitrary precision calculator"
+DESCRIPTION="Calc is an interactive calculator which provides for easy large \
+numeric calculations, but which also can be easily programmed for difficult or \
+long calculations. It can accept a command line argument, in which case it \
+executes that single command and exits. Otherwise, it enters interactive mode. \
+In this mode, it accepts commands one at a time, processes them, and displays \
+the answers. In the simplest case, commands are simply expressions which are \
+evaluated."
+HOMEPAGE="http://www.isthe.com/chongo/tech/comp/calc/"
+COPYRIGHT="1999-2016 Landon Curt Noll"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="http://www.isthe.com/chongo/src/calc/calc-$portVersion.tar.bz2"
+CHECKSUM_SHA256="0fcf60917efb10aab145f1180c278a3ab63d789aad6d5e5f066a4458b5f9b692"
+PATCHES="calc-$portVersion.patchset"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86_gcc2 x86"
+
+PROVIDES="
+	calc$secondaryArchSuffix = $portVersion
+	cmd:calc$secondaryArchSuffix = $portVersion
+	cmd:calc_static$secondaryArchSuffix = $portVersion
+	cmd:cscript$secondaryArchSuffix = $portVersion
+	lib:libcalc$secondaryArchSuffix
+	lib:libcustcalc$secondaryArchSuffix
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libreadline
+	lib:libhistory
+	lib:libncurses
+	"
+
+PROVIDES_devel="
+	calc${secondaryArchSuffix}_devel = $portVersion
+	devel:libcalc$secondaryArchSuffix
+	devel:libcustcalc$secondaryArchSuffix
+	"
+REQUIRES_devel="
+	calc$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libreadline
+	devel:libhistory
+	devel:libncurses
```

Comment:
Well, it looks like your calc_data is ready! You have become an expert!
I would like to apologize, however, because I forgot to mention two things that need to be checked for the calc_data:
1) Before we create such a sub-package, we need to make sure that, without the calc_data, the contents of the files we want to put in calc_data are the same, when we compare what we get for a primary arch build with what we get for a secondary arch build. When we think about this it seems an obvious thing to do, but I should however have mentioned it.
2) The second thing that needs to be checked is that haikuporter is happy with the calc_data stuff not only for a primary arch build, but **_also_** for a secondary arch build. And that's where the problems might come: I remember that I have hit this problem with a package, but I don't remember if it was with file (and then that I found how to fix it) or if it was with another recipe, for which I had to give up the creation of the arch-independent sub-package.

The hardest part of calc_data is to make sure that INSTALL does not fail with secondary arch builds, when it tries to copy the data files to the read-only directory that holds the files from the BUILD_REQUIRE~S~d calc_data. I'm almost sure that we will hit this problem, and that we will need to find a workaround.
So this means it is a bit more complicated to make sure calc_data work perfectly for secondary arch builds.
If your recipe is already working fine with calc_data on secondary arch builds, then we only need to choose a nice text for SUMMARY_data. Otherwise, what I suggest is that you keep your current calc_data changes locally on a safe place and restore, on this pull request, the version without calc_data, and then come back later (if you wish) after you have done as many other tasks as you wish.
Shall we do that?
BTW, as I have already said, your recipe is already great without calc_data, and you have done a great job!
Let me just check that the READLINE stuff is working fine and then we can merge this recipe and validate your task so that you can enjoy other tasks ;-)